lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -71,6 +141,78 @@ class ComponentLoweringState {
     opNames[op] = getUniqueName(prefix);
   }
 
+  template <typename TLibraryOp>
+  TLibraryOp getNewLibraryOpInstance(PatternRewriter &rewriter, Location loc,
+                                     TypeRange resTypes) {
+    IRRewriter::InsertionGuard guard(rewriter);
+    rewriter.setInsertionPoint(compOp.getBody(), compOp.getBody()->begin());
+    auto name = TLibraryOp::getOperationName().split(".").second;
+    auto uniqueName = getUniqueName(name);
+    return rewriter.create<TLibraryOp>(loc, rewriter.getStringAttr(uniqueName),
+                                       resTypes);
+  }
+
+  /// Register value v as being evaluated when scheduling group.
+  void registerEvaluatingGroup(Value v, calyx::GroupInterface group) {
+    valueGroupAssigns[v] = group;
+  }
+
+  /// Return the group which evaluates the value v. Optionally, caller may
+  /// specify the expected type of the group.
```

Comment:
This isn't really making sense to me, but maybe when I see it used in the future it will make more sense.

Reply:
This will be used for schedule generation. When the conditional value for `if` and `while` ops are set, we may call this function to retrieve the corresponding combinational group which needs to be scheduled as the 'with` group.